### PR TITLE
Gravatar: Address the Gravatar Quick Editor scenario

### PR DESCRIPTION
**Pull request by wellyshen**
Related to 108240-gh-Automattic/gravatar

## Proposed Changes

* Disable the email input from the magic code login view
* Hide the "Switch email" from the magic code verification view
* Disable the email input from the password login view
* Hide the social login button from the password login view
* Hide the "Create one" option from the footer of the password login view
* Adjust the subtitle from the password login view

## Why are these changes being made?

* Please refer to 108240-gh-Automattic/gravatar

### Preparation

* [Setup Calypso](https://github.com/Automattic/wp-calypso?tab=readme-ov-file#getting-started)
* Check out this PR, and run `yarn start`
* (IMPORTANT) Allow the browsers' 3rd-party cookies (e.g. [Google Chrome](https://support.google.com/accounts/answer/61416?hl=en&co=GENIE.Platform%3DDesktop))
* **Log out your account**

### Test Gravatar Quick Editor Flow

* Modify the `<YOUR_SECONDARY_EMAIL>` part of the following URL and test it:
  * `http://calypso.localhost:3000/log-in/link?email_address=<YOUR_SECONDARY_EMAIL>&gravatar_from=3rd-party&client_id=1854&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3D7cb30c204432e20253ec71dda82f83c915bc16d2b1cb057d131cdf48eaecf958%26redirect_uri%3Dhttps%253A%252F%252Fgravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26from-calypso%3D1`

* The email input should be disabled

截圖 2024-07-04 下午3 48 53

* Click the "Sign in another way", and the password login view should look like this:

截圖 2024-07-04 下午5 51 24

* There's no "Switch email" option on the magic code verification view

截圖 2024-07-04 下午3 49 02

### Other Flows / Products Should Still Work The Same

* Gravatar (Magic)
  * [Dev](http://calypso.localhost:3000/log-in/link?client_id=1854&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3D501c992bba35eec2a189fa5ac204305203ce020aa5a1490255a94cc904783d57%26redirect_uri%3Dhttps%253A%252F%252Fgravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26gravatar_from%3Dsignup%26from-calypso%3D1&gravatar_from=signup) 
  * [Prod](https://wordpress.com/log-in/link?client_id=1854&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3D501c992bba35eec2a189fa5ac204305203ce020aa5a1490255a94cc904783d57%26redirect_uri%3Dhttps%253A%252F%252Fgravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26gravatar_from%3Dsignup%26from-calypso%3D1&gravatar_from=signup)
  
* Gravatar (Password)
  * [Dev](http://calypso.localhost:3000/log-in?redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3D354d13a74d619f5fe347b6dd9ce36dcdedafd3c9ecd820b8fd7435d0e8d7f9a3%26redirect_uri%3Dhttps%253A%252F%252Fgravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26from-calypso%3D1&client_id=1854)
  * [Prod](https://wordpress.com/log-in?redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fclient_id%3D1854%26response_type%3Dcode%26blog_id%3D0%26state%3D354d13a74d619f5fe347b6dd9ce36dcdedafd3c9ecd820b8fd7435d0e8d7f9a3%26redirect_uri%3Dhttps%253A%252F%252Fgravatar.com%252Fconnect%252F%253Faction%253Drequest_access_token%26from-calypso%3D1&client_id=1854)

* WP Job Manager (Magic)
  * [Dev](http://calypso.localhost:3000/log-in/link?client_id=90057&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%2F%3Fclient_id%3D90057%26response_type%3Dcode%26state%3DeyJyZWRpcmVjdF90byI6Imh0dHBzOlwvXC93cGpvYm1hbmFnZXIuY29tXC9teS1hY2NvdW50In0%26redirect_uri%3Dhttps%253A%252F%252Fwpjobmanager.com%252Fwp-json%252Fwpjmcom-auth%252Fv1%252Fcallback%26blog_id%3D0%26from-calypso%3D1)
  * [Prod](https://wordpress.com/log-in/link?client_id=90057&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthenticate%2F%3Fclient_id%3D90057%26response_type%3Dcode%26state%3D01b7d7bfcfd126450053529ce09fefb5%26redirect_uri%3Dhttp%253A%252F%252Fwpjobmanagercom.vipdev.lndo.site%253A8000%252Fwp-json%252Fwpjmcom-auth%252Fv1%252Fcallback&create_account=1)

* WP Job Manager (Password)
  * [Dev](http://calypso.localhost:3000/log-in?redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%2F%3Fclient_id%3D90057%26response_type%3Dcode%26state%3DeyJyZWRpcmVjdF90byI6Imh0dHBzOlwvXC93cGpvYm1hbmFnZXIuY29tXC9teS1hY2NvdW50In0%26redirect_uri%3Dhttps%253A%252F%252Fwpjobmanager.com%252Fwp-json%252Fwpjmcom-auth%252Fv1%252Fcallback%26blog_id%3D0%26from-calypso%3D1&client_id=90057)
  * [Prod](https://wordpress.com/log-in?redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%2F%3Fclient_id%3D90057%26response_type%3Dcode%26state%3DeyJyZWRpcmVjdF90byI6Imh0dHBzOlwvXC93cGpvYm1hbmFnZXIuY29tXC9teS1hY2NvdW50In0%26redirect_uri%3Dhttps%253A%252F%252Fwpjobmanager.com%252Fwp-json%252Fwpjmcom-auth%252Fv1%252Fcallback%26blog_id%3D0%26from-calypso%3D1&client_id=90057)

* WP.com (Magic)
   * [Dev](http://calypso.localhost:3000/log-in/link)
   * [Prod](https://wordpress.com/log-in/link)

* WP.com (Password)
   * [Dev](http://calypso.localhost:3000/log-in)
   * [Prod](https://wordpress.com/log-in)

* Woo (Magic)
  * [Dev](http://calypso.localhost:3000/log-in/link?redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%2F%3Fresponse_type%3Dcode%26client_id%3D50916%26state%3Dfe6b7fe71924cbd378654e8165ab2b6bf0f2d658065cd27bf038faf6b49a8e9a%26redirect_uri%3Dhttps%253A%252F%252Fwoocommerce.com%252Fwc-api%252Fwpcom-signin%253Fnext%253D%25252F%26blog_id%3D0%26wpcom_connect%3D1%26wccom-from%26calypso_env%3Dproduction%26from-calypso%3D1&client_id=50916)
  * [Prod](https://wordpress.com/log-in/link?redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%2F%3Fresponse_type%3Dcode%26client_id%3D50916%26state%3Dfe6b7fe71924cbd378654e8165ab2b6bf0f2d658065cd27bf038faf6b49a8e9a%26redirect_uri%3Dhttps%253A%252F%252Fwoocommerce.com%252Fwc-api%252Fwpcom-signin%253Fnext%253D%25252F%26blog_id%3D0%26wpcom_connect%3D1%26wccom-from%26calypso_env%3Dproduction%26from-calypso%3D1&client_id=50916)

* Woo (Password)
  * [Dev](http://calypso.localhost:3000/log-in?client_id=50916&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%2F%3Fresponse_type%3Dcode%26client_id%3D50916%26state%3De3654aa37611ba4c042460b75d48b651b84adf46fb770a3bc0aba0dfdadd4cb9%26redirect_uri%3Dhttps%253A%252F%252Fwoocommerce.com%252Fwc-api%252Fwpcom-signin%253Fnext%253D%25252F%26blog_id%3D0%26wpcom_connect%3D1%26wccom-from%26calypso_env%3Dproduction%26from-calypso%3D1)
  * [Prod](https://wordpress.com/log-in?client_id=50916&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%2F%3Fresponse_type%3Dcode%26client_id%3D50916%26state%3De3654aa37611ba4c042460b75d48b651b84adf46fb770a3bc0aba0dfdadd4cb9%26redirect_uri%3Dhttps%253A%252F%252Fwoocommerce.com%252Fwc-api%252Fwpcom-signin%253Fnext%253D%25252F%26blog_id%3D0%26wpcom_connect%3D1%26wccom-from%26calypso_env%3Dproduction%26from-calypso%3D1)
    
* Jetpack (Magic)
  * [Dev](http://calypso.localhost:3000/log-in/link?client_id=69040&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fresponse_type%3Dtoken%26client_id%3D69040%26redirect_uri%3Dhttps%253A%252F%252Fcloud.jetpack.com%252Fconnect%252Foauth%252Ftoken%253Fnext%253D%25252Fpricing%25253Futm_source%25253Dgoogle%252526utm_medium%25253Dcpc%252526utm_campaign%25253D20732627996%252526utm_content%25253D592794838992%252526utm_term%25253Djetpack%252526utm_term%25253Djetpack%252526utm_campaign%25253Dgoogle_jetpack_search_BRAND_other_countries%252526utm_source%25253Dadwords%252526utm_medium%25253Dppc%252526hsa_acc%25253D6173802314%252526hsa_cam%25253D20732627996%252526hsa_grp%25253D160816494928%252526hsa_ad%25253D592794838992%252526hsa_src%25253Dg%252526hsa_tgt%25253Dkwd-298987025683%252526hsa_kw%25253Djetpack%252526hsa_mt%25253De%252526hsa_net%25253Dadwords%252526hsa_ver%25253D3%252526gad_source%25253D1%252526gclid%25253DEAIaIQobChMIl_2KwY3qhgMV2sJMAh0OZgfnEAAYASAAEgIDEvD_BwE%26scope%3Dglobal%26blog_id%3D0%26from-calypso%3D1)
  * [Prod](https://wordpress.com/log-in/link?client_id=69040&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fresponse_type%3Dtoken%26client_id%3D69040%26redirect_uri%3Dhttps%253A%252F%252Fcloud.jetpack.com%252Fconnect%252Foauth%252Ftoken%253Fnext%253D%25252Fpricing%25253Futm_source%25253Dgoogle%252526utm_medium%25253Dcpc%252526utm_campaign%25253D20732627996%252526utm_content%25253D592794838992%252526utm_term%25253Djetpack%252526utm_term%25253Djetpack%252526utm_campaign%25253Dgoogle_jetpack_search_BRAND_other_countries%252526utm_source%25253Dadwords%252526utm_medium%25253Dppc%252526hsa_acc%25253D6173802314%252526hsa_cam%25253D20732627996%252526hsa_grp%25253D160816494928%252526hsa_ad%25253D592794838992%252526hsa_src%25253Dg%252526hsa_tgt%25253Dkwd-298987025683%252526hsa_kw%25253Djetpack%252526hsa_mt%25253De%252526hsa_net%25253Dadwords%252526hsa_ver%25253D3%252526gad_source%25253D1%252526gclid%25253DEAIaIQobChMIl_2KwY3qhgMV2sJMAh0OZgfnEAAYASAAEgIDEvD_BwE%26scope%3Dglobal%26blog_id%3D0%26from-calypso%3D1)

* Jetpack (Password)
  * [Dev](http://calypso.localhost:3000/log-in?client_id=69040&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fresponse_type%3Dtoken%26client_id%3D69040%26redirect_uri%3Dhttps%253A%252F%252Fcloud.jetpack.com%252Fconnect%252Foauth%252Ftoken%253Fnext%253D%25252F%26scope%3Dglobal%26blog_id%3D0%26from-calypso%3D1)
  * [Prod](https://wordpress.com/log-in?client_id=69040&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fresponse_type%3Dtoken%26client_id%3D69040%26redirect_uri%3Dhttps%253A%252F%252Fcloud.jetpack.com%252Fconnect%252Foauth%252Ftoken%253Fnext%253D%25252F%26scope%3Dglobal%26blog_id%3D0%26from-calypso%3D1)

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by wellyshen** [comment]
@aaronfc Do you mind taking a look at this PR, thanks!

File: client/blocks/login/index.jsx [hunk 1]
```diff
@@ -575,11 +576,16 @@ class Login extends Component {
...
+					const isFromGravatar3rdPartyApp =
+						isGravatarOAuth2Client( oauth2Client ) && currentQuery?.gravatar_from === '3rd-party';
+
 					postHeader = (
 						<p className="login__header-subtitle">
-							{ translate(
-								'If you prefer logging in with a password, or a social media account, choose below:'
-							) }
+							{ isFromGravatar3rdPartyApp
+								? translate( 'Please log in with your email and password.' )
```
Comment (aaronfc) [inline client/blocks/login/index.jsx:585]: I think this message is wrong. There is no password, right? Maybe this should be "Please log in with your email.". And that's all?

File: client/blocks/login/login-form.jsx [hunk 2]
```diff
@@ -823,6 +829,12 @@ export class LoginForm extends Component {
 			} );
 		}
 
+		const shouldShowSocialLoginForm =
+			config.isEnabled( 'signup/social' ) &&
+			! isFromAutomatticForAgenciesReferralClient &&
+			! isCoreProfilerLostPasswordFlow &&
+			! isFromGravatar3rdPartyApp;
```
Comment (aaronfc) [inline client/blocks/login/login-form.jsx:837]: Nice refactor! ✋

File: client/login/magic-login/index.jsx [hunk 3]
```diff
@@ -691,9 +691,11 @@ class MagicLogin extends Component {
 					{ translate( 'Continue' ) }
 				</FormButton>
 				<footer className="grav-powered-magic-login__footer">
-					<button onClick={ this.handleGravPoweredEmailSwitch }>
-						{ translate( 'Switch email' ) }
-					</button>
+					{ query?.gravatar_from !== '3rd-party' && (
```
Comment (aaronfc) [inline client/login/magic-login/index.jsx]: What do you think about extracting this to `isFromGravatarThirdParty` variable?

File: client/login/magic-login/index.jsx [hunk 4]
```diff
@@ -926,14 +928,20 @@ class MagicLogin extends Component {
...
 		let headerText = translate( 'Sign in with your email' );
+		let isEmailInputDisabled = isRequestingEmail;
 
 		if ( isGravatar ) {
 			headerText =
 				query?.gravatar_from === 'signup'
 					? translate( 'Create your Profile' )
 					: translate( 'Edit your Profile' );
+
+			if ( query?.gravatar_from === '3rd-party' ) {
```
Comment (aaronfc) [inline client/login/magic-login/index.jsx]: Same: Extracting to a `isFromGravatarThirdParty` variable? (for easier reading)

File: client/blocks/login/index.jsx (hunk 1 above)
Comment (aaronfc) [reply client/blocks/login/index.jsx:585]: Nevermind, I just saw that this is only for email/password flow. All good!

**Review by aaronfc — APPROVED** [review]
Looks good and works well! I left some minor comments, but nothing is a blocker.

Good job! ✋

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16076745

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @wellyshen for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16076745).

**Pull request merged**